spec/support/shared_examples/create_service_with_price_rule_shared_examples.rb

```diff
@@ -0,0 +1,21 @@
+RSpec.shared_examples "creates a product with billing mode" do |product_type, billing_mode|
+  let(:logged_in_user) { administrator }
+
+  it "can create a #{product_type}, which automatically creates a price rule" do
+    visit facility_products_path(facility)
+    all("a", text: product_type.capitalize)[0].click
+    click_link "Add #{product_type.capitalize}"
+
+    fill_in "Name", with: "My new Product", match: :first
+    fill_in "URL Name", with: "new-service"
+    select "Required", from: "#{product_type}[user_notes_field_mode]"
+    select billing_mode, from: "#{product_type}[billing_mode]"
+
+    click_button "Create"
+    click_on "Pricing"
+
+    expect(page).to have_content "#{PriceGroup.nonbillable} (#{PriceGroup.nonbillable.type_string}) $0"
```

Comment:
If we're only creating nonbillable, do we need the `billing_mode` passed in?

Reply:
We are not only just creating products with nonbillable billing_mode, we are creating skip review too.
As I understand, creating products of both billing modes generates "nonbillable" price groups (base price group with $0 cost). This code checks that the price group is being displayed correctly in the product's price tab.

`PriceGroup.nonbillable ` returns the base price group

```
  def self.nonbillable
    base
  end

  def self.base
    globals.find_by(name: Settings.price_group.name.base)
  end
```